src/idl_gen_dart.cpp

```diff
@@ -281,7 +281,15 @@ class DartGenerator : public BaseGenerator {
             enum_type + "Reader();\n\n";
     code += "  @override\n";
     code += "  String toString() {\n";
-    code += "    return '" + enum_type + "{value: $value}';\n";
+    code += "    switch (value) {\n";
+    for (auto it = enum_def.Vals().begin(); it != enum_def.Vals().end(); ++it) {
+      auto &ev = **it;
+      const auto enum_var = namer_.Variant(ev);
+      code += "      case " + enum_def.ToString(ev) + ": return \"" + enum_var +
+              "\";\n";
+    }
+    code += "      default: return \"\";\n";
+    code += "    }\n";
```

Comment:
The proposed format loses the information about the containing enum. What's the reasoning behind that proposal?

Also, if we should change this, i'd suggest aligning with dart Enum toString() instead:

```
enum Color { red, green, blue }

main() {
  test('test', () {
    expect(Color.red.toString(), 'Color.red');
  });
}
```

Additionally, that would seem like a good time to switch to actual enums. Do you think we'd be able to do all that's needed to construct in the scope of flatbuffers if we use enhanced enum syntax?

Reply:
Update: I've looked into enhanced enums and it works fine, even with the default toString. I'll prep a PR.

Reply:
@vaind It seems that there will also be issues with #5467

Reply:
Current code already handles that and throws in such a case: `throw StateError('Invalid value $value for bit flag enum Abc');`. Whether that's the best thing to do is a different topic.

the behavior remains unchanged after the change to enhanced enums (https://github.com/google/flatbuffers/pull/8313)

Reply:
The issue is not whether users can handle it, but rather compatibility. My opinion is that when users develop software, they write enumeration processing code according to normal logic (catch may not be written at this time). When an unrecognized enumeration value is passed in externally, an error will be reported, which means compatibility is reduced (at least users need to consider more things during initial development, which will bring more mental burden). Therefore, I believe that such modifications require more people to participate and more discussion.

Reply:
See what others think. If the new PR is merged, I will close this PR